Uyanda.Coffee.Persistence/Accessors/BeverageAccessor.cs

```diff
@@ -65,61 +67,190 @@ public async Task<InvoiceModel> PurchaseAsync(IEnumerable<LineItemModel> lineIte
+
+
         }
 
-        public async Task<BeverageSizeCostModel> UpsertBeverageSizeCostAsync(BeverageSizeCostModel price)
+
+        private async Task<InvoiceModel> NormalPurchaseAsync(IEnumerable<LineItemModel> lineItems, CustomerModel customer)
         {
-          //update price
+            var beveragePrices = await localDbContext.BeverageCost.AsNoTracking()
+                .Select(c => new { c.Id, c.Cost })
+                .ToDictionaryAsync(item => item.Id, item => item.Cost);
 
-            
-                
-            if (price.Id > 0)
+            var totalCost = lineItems
+                .Sum(c => c.Count * beveragePrices[c.BeverageSizeCostId]);
+
+            int discount = 10;
+
+            var earnedPoints = totalCost / discount;
+
+            var isUserFound = await localDbContext.Customer
+                    .Where(c => c.Id == customer.Id)
+                    .AnyAsync();
+
+            var costPerItem = await localDbContext.BeverageCost.AsNoTracking()
+                .Select(c => new { c.Id, c.Cost }).ToDictionaryAsync(item => item.Id, item => item.Cost);
+
+            var purchase = lineItems
+                .Select(c => new LineItemModel
+                {
+                    BeverageSizeCostId = c.BeverageSizeCostId,
+                    Count = c.Count,
+                    CostPerItem = costPerItem[c.BeverageSizeCostId]
+                }).ToList();
+
+            if (!isUserFound)
+            {
+                if (customer.Id != 0)
+                    throw new InvalidOperationException("Invalid Id value");
+
+                var customerInvoice = new InvoiceModel
+                { 
+                    Date = DateTime.Now,
+                    LineItems = purchase,
+                    CustomerId = 1
+                };
+
+                await localDbContext.AddAsync(ToEntity(customerInvoice));
+
+                await localDbContext.SaveChangesAsync();
+
+                return customerInvoice;
+            }        
+
+            if (isUserFound)
             {
-                var beveragePrice = await localDbContext.BeverageCost
-                    .Where(c => c.Id == price.Id)
-                    .Include(d => d.Beverage)
-                    .SingleAsync();
+                if (customer.Id == 1)
+                    throw new InvalidOperationException("invalid ID");
+
+                customer.Points += earnedPoints;
 
-                beveragePrice.Beverage.Name = price.Beverage.Name;
+                await localDbContext.AddAsync(ToEntity(customer));
 
-                beveragePrice.Cost = price.Cost;
+                await localDbContext.SaveChangesAsync();
 
-                beveragePrice.Beverage.IsActive = price.Beverage.IsActive;
+                var customerInvoice = new InvoiceModel
+                {
+                    Date = DateTime.Now,
+                    LineItems = purchase,
+                    CustomerId = customer.Id
+                };
```

Comment:
Please store _IsRedeemingPoints_, _DiscountedPoints_ and _FinalInvoicePrice_ from in the DB.